### ISSUE TITLE
micro-ros  host build problem 

### ISSUE THREAD

**Issue by DorBenHarush** [micro-ROS/micro_ros_setup#184]
Hi, 
I`m trying to create firmware for host and some unexpected errors occurred during the build process.
These are the errors:

![Screenshot from 2020-08-27 10-51-00](https://user-images.githubusercontent.com/69308426/91413810-550c7c80-e854-11ea-8fc8-710bbf490618.png)

I followed the tutorial [here](https://micro-ros.github.io/docs/tutorials/core/first_application_linux/), and I`m using a docker container as shown. 

Regards,
Dor.

**Comment by pablogs9** [comment]
Hello @DorBenHarush we have found some issues related to the build system thanks to your issue.

We have working here: https://github.com/micro-ROS/micro_ros_setup/pull/185, now it passing the CI test and once it is merged you can try again the tutorial without (hopefully) any issue.

Thanks!!

**Comment by pablogs9** [comment]
The PR has been merged and the issue solved. Please @DorBenHarush, reopen this issue if this does not solve your problem.

Sorry for the inconvenience.

**Comment by DorBenHarush** [comment]
Thanks, it worked. But I think the tutorial is outdated.
I typed this:
ros2 run micro_ros_agent micro_ros_agent udp4 --port 8888
as shown [here](https://micro-ros.github.io//docs/tutorials/core/first_application_linux/)
and recived:
`Usage: 'MicroXRCEAgent <udp4|udp6|tcp4|tpc6|serial|pseudoterminal> <<args>>'
For a more detailed description about all the available arguments, please execute the agent with '-h/--help' option.`
Can you please guide me. 

Regards, 
Dor.

**Comment by pablogs9** [comment]
@jamoralp can you check this? 

Related: https://answers.ros.org/question/360494/micro-ros-agent-displays-usage-and-wont-start-with-valid-arguments/

**Comment by pablogs9** [comment]
Hello @DorBenHarush  we have solved it here: https://github.com/eProsima/Micro-XRCE-DDS-Agent/pull/183

Can you check if the Agent is working now and close the issue?

Thanks!!

**Comment by DorBenHarush** [comment]
Hi, the Agent runs but when I try to run the micro-ROS node with `ros2 run micro_ros_demos_rclc ping_pong` this response pops:
`Package 'micro_ros_demos_rclc' not found`. I checked the pkg list and the package is not there.

**Comment by pablogs9** [comment]
Hello @DorBenHarush , `micro_ros_demos_rclc` package should be found inside [this folder](https://github.com/micro-ROS/micro-ROS-demos/tree/foxy/rclc/ping_pong) can you check if you have this folder inside `src`?

Have you done the `source install/local_setup.bash`? Anyways I'm doing checks for this tutorial step by step, I will tell you in a while.

**Comment by DorBenHarush** [comment]
Hi @pablogs9 , I typed `ros2 pkg list | grep micro` to check if the package `micro_ros_demos_rclc` exists. furthermore I checked in the `src` folder and it only contained  `micro_ros_build`.

**Comment by pablogs9** [comment]
Can you paste here a step by step procedure with all commands that you are executing for reproducing this issue? I would like to check your workflow 

Thanks!

**Comment by DorBenHarush** [comment]
I have solved the problem, since I'm working from docker container I needed to create another container in roder to run the `micro_ros_deomos_rclc`. Then I repeated the first stages in the tutorial to recreate the enviorment. My mistake was **this**: I didnt proceed to the `create_firmware_ws.sh` step in the new container, Therefore I didnt download `micro_ros_demos_rclc` package. 

I have tried it again but this time, instead of creating another container, I created another session to the same container and it worked. I think this tutorial needs some rewriting in order to be more clear.
Thank you for your support.
Regards,
Dor.
